### PR TITLE
Attempted to make tests pass in any time zone

### PR DESCRIPTION
**Pull request by fearphage**
- added test section labels

@mantoni Here's an attempt at a universal fix.

### CODE REVIEW

File: test/suite.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (mantoni) [inline test/suite.sh]: Oh yes, that is better 👍

File: test/suite.sh [hunk 2]
```diff
@@ -13,6 +13,7 @@ assert "node bin/cmd.js < test/async-long.js" "1
 3
 4
 5"
+assert_end basic
```
Comment (mantoni) [inline test/suite.sh]: I didn't know about this feature. Very nice with a test suite that takes this long to run. Thanks!

File: test/suite.sh [hunk 3]
```diff
@@ -37,8 +40,10 @@ assert "node_modules/.bin/browserify --debug test/sourcemaps-console.js | node b
 assert "node_modules/.bin/browserify --debug test/sourcemaps-uncaught.js | node bin/cmd.js | head -n 2" "Error: oups
       at test/sourcemaps-uncaught.js:2"
 
-PHANTOM=`which phantomjs`
-NODE=`which node`
+assert_end debugging
+
+PHANTOM=$(which phantomjs)
+NODE=$(which node)
```
Comment (mantoni) [inline test/suite.sh]: Why is it better to do it this way? Because it doesn't fork a sub-shell?

**Comment by mantoni** [comment]
Thanks! Unfortunately, date cannot be used this way on OS X. This is what the build says here:

```
all 4 basic tests passed in 11.000s.
all 6 error tests passed in 12.000s.
test #3 "node_modules/.bin/browserify --debug test/sourcemaps-console.js | node bin/cmd.js" failed:
        expected "      at test/sourcemaps-console.js:3"
        got "      s node_modules/browser-pack/_prelude.js:1"
1 of 4 debugging tests failed in 11.000s.
all 5 cli flag tests passed in 11.000s.
date: illegal time format
usage: date [-jnu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ...
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
date: illegal time format
usage: date [-jnu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ...
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
test/suite.sh: line 65: ( - ) / 60: syntax error: operand expected (error token is ") / 60")
date: illegal time format
usage: date [-jnu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ...
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
test/suite.sh: line 68: ( - ) / 60: syntax error: operand expected (error token is ") / 60")
test #2 "node bin/cmd.js < test/timezone.js" failed:
        expected nothing
        got "-120"
test #3 "node bin/cmd.js < test/timezone.js" failed:
        expected nothing
        got "240"
2 of 3 miscellaneous tests failed in 8.000s.
make: *** [default] Error 1
```

**Comment by mantoni** [comment]
At first, I was thinking of using [dst](https://www.npmjs.com/package/dst), but then I went with the plain bash solution. Was this still not working on your end?

File: test/suite.sh (hunk 3 above)
Comment (fearphage) [inline test/suite.sh]: Well the biggest advantage I know is that you can use it anywhere. Like you can `$(nest $(commands $(deeply)))` without any problems. You can't nest back ticks. Also back ticks are deprecated.

**Comment by fearphage** [comment]
It was working, but only for me. You're on a mac I presume?

**Comment by mantoni** [comment]
Yes, I'm on a Mac. It worked for me also before you changed it. Why did you change it then?

File: test/suite.sh (hunk 3 above)
Comment (mantoni) [inline test/suite.sh]: Thanks for explaining. I should change my habits then 💣💥

**Comment by mantoni** [comment]
Sorry, I got your comment wrong. My actual question is: Why did my fix not work for you?

**Comment by fearphage** [comment]
Your fix worked for me and everyone in my timezone. It didn't work anyone outside of our 2 timezones. I was trying to make a generic fix instead of hard code it. I'll figure it out when I get a chance.

**Comment by mantoni** [comment]
Ah really? I thought the issue was just the daylight saving 😟

**Comment by mantoni** [comment]
This PR now has conflicts. @fearphage Sorry, I completely forgot about this one. Is it still relevant?

**Comment by fearphage** [comment]
I think there is valuable clean up here even if you don't take the time zone pieces.

**Comment by fearphage** [comment]
@mantoni Pulled in master and fixed conflicts. I reverted the time-based changes. I also fixed the commented out test. The problem was that you were killing the `$PATH`, but `env node` (at the top of `bin/cmd.js` and `./node_modules/.bin/phantomjs`) both rely on the path to find node. I set the path to only point to the node dir. I don't really see what the goal of the test, but it works now.

One test is failing though. I didn't attemp to fix it.

```shell
➜  phantomic git:(fix-timezone-tests) ✗ ./test/suite.sh                                                                                                                            
all 4 basic tests passed in 4.182s.
all 6 error tests passed in 4.876s.
test #3 "node_modules/.bin/browserify --debug test/sourcemaps-console.js | node bin/cmd.js" failed:
        expected "      at test/sourcemaps-console.js:3"
        got "      s node_modules/browser-pack/_prelude.js:1"
1 of 4 debugging tests failed in 3.596s.
all 7 cli flag tests passed in 5.700s.
all 3 miscellaneous tests passed in 2.833s.
all 2 JS error tests passed in 1.806s.
```

**Comment by mantoni** [comment]
@fearphage Thanks a lot for cleaning all this up. These are good improvements.

The failing test fails for me as well. I think it used to work with Phantom 1.9 and stopped working with v2. I didn't investigate yet.

The test you fixed is supposed to verify that the `--phantomjs` switch works. The idea was to make sure the binary is not in the `PATH` and specify the binary explicitly.

**Pull request merged**